### ISSUE TITLE
Proposal: Cannot use X  as Y because the name is already in use

### ISSUE THREAD

**Issue by veewee** [kalessil/phpinspectionsea#452]
When working in an environment with Zend Framework 2 (or >), it is possible that you get following error:

```
E_COMPILE_ERROR: Cannot use X  as Y because the name is already in use
```

This is when you have following sturcture:
```
src/SomeModule/Service
├── SomeService.php
├── SomeOtherService.php
└──Factory
   ├── SomeService.php
   └── SomeOtherService.php
```

And the factory looks like this:

```php
<?php

declare(strict_types=1);

namespace SomeModule\Service\Factory;

use SomeModule\Service\SomeOtherService;
use Zend\ServiceManager\FactoryInterface;
use Zend\ServiceManager\ServiceLocatorInterface;

class PlaceOrderService implements FactoryInterface
{
    public function createService(ServiceLocatorInterface $serviceLocator)
    {
        return new \Order\SomeModule\SomeService(
            $serviceLocator->get(SomeOtherService::class),
        );
    }
}

```

As you can see, the SomeOtherService is refering to `SomeModule\Service\SomeOtherService`. Since there is also a factory named SomeOtherService, this code gives you the error as pasted above.
In this case you should use an alias or a FQCN instead.

There is 1 small problem:  when you have a FQCN, there is an inspection that allows you to import the class. This results in the exact same error as mentioned before.

**Comment by funivan** [comment]
I find `Unnecessary full qualified name` inspection under the `PHP -> Code style` group. This inspection fires an error ?

**Comment by veewee** [comment]
In this case it is not an Unnecesarry FQN because it conflicts with a short classname in the same namespace. So you should be able to choose an alias or allow FQN. 

This ticket was about adding an inspection for the issue 'E_COMPILE_ERROR: Cannot use X  as Y because the name is already in use' when the IDE did shorten the classname which results in a conflict.

**Comment by funivan** [comment]
```php
  namespace A\Library {
    class Session {    }
  }
  namespace My\Application {
    class Session {    }
  }
  namespace My\Application {
    use A\Library\Session;
    class Test {
      public function make() {
        new Session();
      }
    }
  }
```
I get an error from the PhpStorm inspection: `Can not use Session because name is already in use`
![11 07 16](https://user-images.githubusercontent.com/425208/29062290-c84dd36e-7c29-11e7-8490-6cc6a3d46dcb.png)

**Comment by veewee** [comment]
@funivan. Indeed, that seems to work here as well.
Strange that I did not retrieve information in my codebase. Might be a bug in phpstorm...

Thanks for looking into it! Let's close this for now.
